### PR TITLE
Feature/kor 48 load data to db additional

### PR DESCRIPTION
**Pull request by greenhorn31**

### CODE REVIEW

File: app/src/main/java/com/leoart/koreanphrasebook/data/network/firebase/phrases/PhrasesRequest.kt [hunk 1]
```diff
@@ -25,29 +26,37 @@ class PhrasesRequest : FireBaseRequest() {
         }
     }
 
-
-    fun getPhrases(categoryName: String): Observable<List<Phrase>> {
-        return Observable.create({ subscriber ->
-            dataBase.reference?.child("$CATEGORY_PHRASES/$categoryName")?.addListenerForSingleValueEvent(object : ValueEventListener {
+    fun getPhrases(): Observable<List<EPhrase>> {
```
Comment (khrystynapochynok) [inline app/src/main/java/com/leoart/koreanphrasebook/data/network/firebase/phrases/PhrasesRequest.kt]: @greenhorn31 why request class returns local db entity?

File: app/src/main/java/com/leoart/koreanphrasebook/data/network/firebase/phrases/PhrasesRequest.kt [hunk 2]
```diff
@@ -25,29 +26,37 @@ class PhrasesRequest : FireBaseRequest() {
...
                     }
                     subscriber.onNext(phrases)
                     subscriber.onComplete()
-
                 }
             })
         })
     }
+
+    private fun parseCategoryPhrases(dataSnapshot: DataSnapshot) : List<EPhrase>{
```
Comment (khrystynapochynok) [inline app/src/main/java/com/leoart/koreanphrasebook/data/network/firebase/phrases/PhrasesRequest.kt]: @greenhorn31 move it to repository class

File: app/src/main/java/com/leoart/koreanphrasebook/data/repository/AlphabetRepository.kt [hunk 3]
```diff
@@ -26,8 +24,9 @@ class AlphabetRepository(private val context: Context) : CachedRepository<ELette
 
     override fun requestFromNetwork() {
         AlphabetRequest().fetchAlphabet()
-                .subscribeOn(Schedulers.io())
+                .observeOn(Schedulers.io())
                 .subscribe {
+                    clearDB()
```
Comment (khrystynapochynok) [inline app/src/main/java/com/leoart/koreanphrasebook/data/repository/AlphabetRepository.kt]: @greenhorn31 check if data from network is not empty before clearing db

File: app/src/main/java/com/leoart/koreanphrasebook/ui/splash/SplashPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.leoart.koreanphrasebook.ui.splash
+
+class SplashPresenter(){
```
Comment (khrystynapochynok) [inline app/src/main/java/com/leoart/koreanphrasebook/ui/splash/SplashPresenter.kt]: @greenhorn31 why is it empty?

File: app/src/main/java/com/leoart/koreanphrasebook/ui/splash/SplashActivity.kt [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    private lateinit var model: SplashViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_splash)
+        model = ViewModelProviders.of(
+                this,
+                ViewModelFactory(applicationContext)
+        ).get(SplashViewModel::class.java)
+        model.openNextScreen()
```
Comment (khrystynapochynok) [inline app/src/main/java/com/leoart/koreanphrasebook/ui/splash/SplashActivity.kt]: @greenhorn31 this is wrong ViewModel implementation. You should subscribe to data change, not to request response. Please check https://developer.android.com/topic/libraries/architecture/viewmodel for more information.

File: app/src/main/java/com/leoart/koreanphrasebook/ui/splash/SyncDataRepository.kt [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        }
+    }
+
+    fun refreshDB(): Completable {
+        val refreshChain = repositories.map { it.refreshData() }
+        return Completable.merge(refreshChain)
+    }
+
+    private fun sendRefreshError() {
+        // todo
```
Comment (khrystynapochynok) [inline app/src/main/java/com/leoart/koreanphrasebook/ui/splash/SyncDataRepository.kt]: @greenhorn31 why empty?

File: app/src/main/java/com/leoart/koreanphrasebook/ui/sync/SyncActivity.kt [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+    private fun initUI() {
+        dismissButton.setOnClickListener {
+            openMainScreen()
+        }
+        syncButton.setOnClickListener {
+            showLoading()
+            dismissButton.isClickable = false
+            syncButton.isClickable = false
+            model.refreshData().observeOn(Schedulers.io())
```
Comment (khrystynapochynok) [inline app/src/main/java/com/leoart/koreanphrasebook/ui/sync/SyncActivity.kt]: @greenhorn31 wrong implementation - https://developer.android.com/topic/libraries/architecture/viewmodel

**Review by khrystynapochynok — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/leoart/koreanphrasebook/data/network/firebase/phrases/PhrasesRequest.kt (hunk 1 above)
Comment (greenhorn31) [reply app/src/main/java/com/leoart/koreanphrasebook/data/network/firebase/phrases/PhrasesRequest.kt]: i dont know, it was changed by Lubomur

**Pull request merged**